x/staking/keeper/grpc_query.go

```diff
@@ -229,7 +225,7 @@ func (k Querier) Delegation(ctx context.Context, req *types.QueryDelegationReque
 		return nil, err
 	}
 
-	delegation, err := k.GetDelegation(ctx, delAddr, valAddr)
+	delegation, err := k.Delegations.Get(ctx, collections.Join(sdk.AccAddress(delAddr), sdk.ValAddress(valAddr)))
```

Comment:
We should have two conditions, one for not found (err is collections.ErrNotFound) and one for other errors